### DISCUSSION TITLE
FileUpload - Convert Extension and Compressing

### DISCUSSION THREAD

**Discussion by berdancanyet** [filamentphp/filament#9095] [Ideas]
I would love to change the image file's extension and reducing its quality by simply using like this:

```
FileUpload::make('detail_image_path')
                                    ->maxSize(1024)
                                    ->compress(0.9)
                                    ->convertTo('.webp')
```

**Comment by ufukseyithan** [comment]
This would be a great feature!

**Comment by joshembling** [comment]
Hi @berdancanyet @ufukseyithan 

I have a plugin created for exactly this feature 🎉 It works with both `FileUpload` and `SpatieMediaLibraryFileUpload`.

Let me know if you have any questions. I hope you get good use out of it! 

https://github.com/joshembling/image-optimizer

`composer require joshembling/image-optimizer`

**Comment by berdancanyet** [comment]
thanks for letting us know! great job! @joshembling
